### PR TITLE
Ignore both swSrc and swDest when globbing for the manifest

### LINKED ISSUE #2229: Ensure that swSrc is automatically added to globIgnores
**Library Affected**:
*workbox-build*

When using `workbox-build` in `injectManifest` mode, `swDest` is automatically added to the `globIgnores` config. We should do this for `swSrc` as well.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @philipwalton

Fixes #2229

We previously had some code that ensured that `swDest` was omitted from the `glob` matches when generating the precache manifest.

This extends that to also ignore `swSrc` in `injectManifest` mode.

I also improved the that generates the `globIgnores` value so that instead of just matching anything that has the same basename as `swSrc`/`swDest`, it actually uses a full path to the file, relative to `globDirectory`. This should reduce the chance of a false positive.

### CODE REVIEW

**Comment by jeffposnick** [comment]
Actually, let me extend this to ignore the `workbox-*.js` runtime files that are generated, too.

File: packages/workbox-build/src/lib/rebase-path.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+  Copyright 2018 Google LLC
```
Comment (philipwalton) [inline packages/workbox-build/src/lib/rebase-path.js]: s/2018/2019/

File: packages/workbox-build/src/lib/rebase-path.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+const upath = require('upath');
+
+module.exports = ({baseDirectory, file}) => {
+  // file starts our relative to the current directory, so make it absolute.
```
Comment (philipwalton) [inline packages/workbox-build/src/lib/rebase-path.js]: s/file/File/

I'm not really sure what you mean by "file starts". Maybe reword as "Files are relative to the..."?

Comment (jeffposnick) [reply packages/workbox-build/src/lib/rebase-path.js]: Sorry, that should have been "out" instead of "our".

But I'll rephrase.
